app/models/downloaded_file.rb

```diff
@@ -0,0 +1,30 @@
+require 'rmt/checksum_verifier'
+
+class DownloadedFile < ApplicationRecord
+
+  def self.add_file!(checksum_type, checksum, local_path)
+    return nil unless local_path.match?(/\.(rpm|drpm)$/)
+    file_size = File.size(local_path)
+    DownloadedFile.create({ checksum_type: checksum_type,
+                            checksum: checksum,
+                            local_path: local_path,
+                            file_size: file_size })
+  rescue StandardError => e
+    # de-duplication is not critical, just move on
+    logger.error e.message
+    logger.error e.backtrace.each { |line| logger.error line }
```

Comment:
So far everywhere else in the code we are not showing stack traces unless debug option is specified.